ui/src/app/views/registration-dashboard.tsx

```diff
@@ -116,18 +119,23 @@ export const RegistrationTasks = [
     description: 'This data use agreement describes how All of Us ' +
       'Research Program data can and cannot be used',
     buttonText: 'View & Sign',
+    featureFlag: serverConfigStore.getValue().enableDataUseAgreement,
     completedText: 'Signed',
     isComplete: (profile: Profile) => {
       return !!profile.dataUseAgreementCompletionTime || !!profile.dataUseAgreementBypassTime;
     },
     onClick: () => navigate(['data-use-agreement'])
   }
-];
+] : (() => {
+  throw new Error('Cannot load registration tasks before config loaded');
+})();
 
-export const RegistrationTasksMap = RegistrationTasks.reduce((acc, curr) => {
-  acc[curr.key] = curr;
-  return acc;
-}, {});
+export const getRegistrationTasksMap = () => getRegistrationTasks()
```

Comment:
I like the idea of using the feature flag to avoid adding boilerplate values to the rest of the tasks.

One note about the api change, `getRegistrationTasks` and `getRegistrationTasksMap` now return different values which I think is a little confusing. It should return the same values but in different data structures (list vs map).

In order to do that, I think we will need to define the list outside of the `getRegistrationTasks` or `getRegistrationTasksMap` methods. Another function will work but if we want to keep it as a compiled array, we can set `featureFlag` to the feature flag key or even a method that checks the store. We can still keep the logic of ignoring undefined values to avoid adding it to other tasks.